src/OrchardCore/OrchardCore.Media.Abstractions/Events/IMediaEventHandler.cs

```diff
@@ -0,0 +1,18 @@
+using System.Threading.Tasks;
+
+namespace OrchardCore.Media.Events
+{
+    /// <summary>
+    /// Event handler fired during operations for existing media.
+    /// </summary>
+    public interface IMediaEventHandler
+    {
+        Task MediaDeletingFileAsync(MediaDeleteContext context);
+        Task MediaDeletedFileSuccessAsync(MediaDeleteContext context);
+        Task MediaDeletingFileFailureAsync(MediaDeleteContext context);
+        Task MediaDeletingDirectoryAsync(MediaDeleteContext context);
+        Task MediaDeletedDirectorySuccessAsync(MediaDeleteContext context);
+        Task MediaDeletingDirectoryFailureAsync(MediaDeleteContext context);
```

Comment:
How about merging DeleteFileSuccess/Failure (and DeleteDirectorySuccess/Failure) into one method and set success/failure flag/bool in MediaDeleteContext object?

Reply:
Yeah that's a much better idea. I'm not that convinced on the need for them at all, so let's wrap them up into one.

Reply:
Done